src/v/kafka/types.h

```diff
@@ -91,6 +92,33 @@ using leader_epoch = named_type<int32_t, struct leader_epoch_tag>;
  */
 static constexpr leader_epoch invalid_leader_epoch(-1);
 
+struct uuid {
+    static constexpr auto length = 16;
+
+    template<typename Consumer>
+    void set_uuid(Consumer& c) {
+        c.consume(length, [this](const char* src, size_t max) {
+            std::copy_n(src, max, _uuid.data());
```

Comment:
Standard question here on why this copy is safe against overruns. Basically, `size_t max` is caller-enforced--parsing a UUID will fail if there is a length mismatch. Correct me if I'm wrong, of course.

I can't yet comment on whether this `set_uuid` implementation is idiomatic--some other core C++ folks may have comments.

Reply:
`c.consume` throws if `length` bytes are not in the buffer. So you'd be safe to call `copy_n` , otherwise an exception would be thrown by `consume` and kafka request dropped

Reply:
Thanks for the feedback, yeah would be interested in knowing how others would create an interface for `set_uuid`

Reply:
It feels a bit awkward to have a consumer here, that's usually a more streamable interface.which is overkill for 16bytes.